packages/server/src/core/credit_transfer_graph.rs

```diff
@@ -0,0 +1,51 @@
+use petgraph::algo::toposort;
+use petgraph::Graph;
+
+use crate::error::AppError;
+use crate::resources::catalog::Catalog;
+use crate::resources::course::CourseBank;
+
+use super::messages;
+use super::types::CreditOverflow;
+
+fn build_credit_transfer_graph(
+    course_banks: &[CourseBank],
+    credit_overflow_rules: &[CreditOverflow],
+) -> Graph<String, ()> {
+    let mut g = Graph::<String, ()>::new();
+    for course_bank in course_banks {
+        g.add_node(course_bank.name.clone());
+    }
+    for credit_rule in credit_overflow_rules {
+        g.add_edge(
+            // unwrap cannot fail because the credit rules are taken from bank names
```

Comment:
Since `validate_acyclic_credit_transfer_graph` is now fallible, please create an appropriate `AppError` for this situation and use the `?` operator instead of unwrapping.

Even though these errors are not supposed to ever happen, it is better to express it in the code instead of in a comment.

Reply:
Can you elaborate what do you mean?
This function return ```Result``` and do handle a case of failing.
Where should I add an appropriate ```AppError```?